core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -312,25 +312,30 @@ public int hashCode() {
         return nodeId.hashCode();
     }
 
+    private String toString;
```

Comment:
> The vast majority of node *objects* will never have the toString called on them more than once (and likely not even once, since DiscoveryNode(s) objects will be [replaced])(https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/cluster/AbstractDiffable.java#L37) on every change.

@bleskes The key piece is the piece about the diffs which I had not realized before. I think this change should be reverted!